### ISSUE TITLE
Cura 3.3 and 3.3.1 fail to start (OSX 10.13.4)

### ISSUE THREAD

**Issue by jimhigson** [Ultimaker/Cura#3750]
3.2 was starting fine.

After upgrading to either 3.3 or 3.3.1 always get this message:

![image](https://user-images.githubusercontent.com/136306/39410660-d22fd4a0-4bf3-11e8-85a3-ddfd42714e23.png)

![image](https://user-images.githubusercontent.com/136306/39410665-f1c61a54-4bf3-11e8-90b9-e5d9b0c49d33.png)

It looks like 3.3 can't read the 3.2 config for materials. The most interesting bit seems to be this:

```
packages/cura/Machines/Models/BrandMaterialsModel.py", line 124, in 
AttributeError: 'NoneType' object has no attribute 'upper'
```

I can send my config files in if that is helpful.

Error traceback:
----------------

Traceback (most recent call last):
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/cura/Machines/Models/BrandMaterialsModel.py", line 58, in __init__
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/cura/Machines/Models/BrandMaterialsModel.py", line 124, in _update
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/cura/Machines/Models/BrandMaterialsModel.py", line 124, in <lambda>
AttributeError: 'NoneType' object has no attribute 'upper'

Full log:
--------

Thread 0x0000700006209000 (most recent call first):
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/Backend/Backend.py", line 155 in _storeStderrToLogThread
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 862 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x0000700005d06000 (most recent call first):
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/Backend/Backend.py", line 147 in _storeOutputToLogThread
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 862 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x0000700005780000 (most recent call first):
  File "/private/var/folders/35/pfk85v5x1td46k__w_657mfw0000gn/T/AppTranslocation/FA77FEE5-5356-4C81-AFA1-A94689E0E1DE/d/Ultimaker Cura.app/Contents/MacOS/plugins/plugins/USBPrinting/USBPrinterOutputDeviceManager.py", line 73 in _updateThread
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 862 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x000070000527d000 (most recent call first):
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 293 in wait
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 424 in acquire
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 86 in _nextJob
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 116 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x0000700004d7a000 (most recent call first):
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 293 in wait
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 424 in acquire
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 86 in _nextJob
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 116 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x0000700004877000 (most recent call first):
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 293 in wait
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 424 in acquire
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 86 in _nextJob
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 116 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x0000700004374000 (most recent call first):
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 293 in wait
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 424 in acquire
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 86 in _nextJob
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 116 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x0000700003e71000 (most recent call first):
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 293 in wait
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 424 in acquire
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 86 in _nextJob
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 116 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x000070000396e000 (most recent call first):
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 293 in wait
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 424 in acquire
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 86 in _nextJob
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 116 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x000070000346b000 (most recent call first):
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 293 in wait
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 424 in acquire
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 86 in _nextJob
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 116 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x0000700002f68000 (most recent call first):
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 293 in wait
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 424 in acquire
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 86 in _nextJob
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/JobQueue.py", line 116 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x0000700002a65000 (most recent call first):
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 297 in wait
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/site-packages/zeroconf.py", line 1569 in wait
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/site-packages/zeroconf.py", line 1153 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x0000700002562000 (most recent call first):
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 297 in wait
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/site-packages/zeroconf.py", line 1569 in wait
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/site-packages/zeroconf.py", line 1017 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x000070000205f000 (most recent call first):
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/site-packages/zeroconf.py", line 946 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x0000700001b5c000 (most recent call first):
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 297 in wait
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 549 in wait
  File "/private/var/folders/35/pfk85v5x1td46k__w_657mfw0000gn/T/AppTranslocation/FA77FEE5-5356-4C81-AFA1-A94689E0E1DE/d/Ultimaker Cura.app/Contents/MacOS/plugins/plugins/UM3NetworkPrinting/UM3OutputDevicePlugin.py", line 274 in _handleOnServiceChangedRequests
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 862 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Thread 0x00007000015d6000 (most recent call first):
  File "/private/var/folders/35/pfk85v5x1td46k__w_657mfw0000gn/T/AppTranslocation/FA77FEE5-5356-4C81-AFA1-A94689E0E1DE/d/Ultimaker Cura.app/Contents/MacOS/plugins/plugins/RemovableDriveOutputDevice/RemovableDrivePlugin.py", line 61 in _updateThread
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 862 in run
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 914 in _bootstrap_inner
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/threading.py", line 882 in _bootstrap

Current thread 0x00007fff91d8f380 (most recent call first):
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/cura/CrashHandler.py", line 364 in _logInfoWidget
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/cura/CrashHandler.py", line 214 in _createDialog
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/cura/CrashHandler.py", line 82 in __init__
  File "/Users/ultimaker/build/3.3/build/inst/bin/cura_app.py", line 101 in exceptHook
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/Qt/Bindings/ControllerProxy.py", line 98 in _onActiveStageChanged
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/Signal.py", line 315 in __performEmit
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/Signal.py", line 212 in emit
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/UM/Controller.py", line 162 in setActiveStage
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/cura/CuraApplication.py", line 768 in runWithGUI
  File "/Users/ultimaker/build/3.3/build/inst/lib/python3.5/site-packages/cura/CuraApplication.py", line 687 in run
  File "/Users/ultimaker/build/3.3/build/inst/bin/cura_app.py", line 141 in <module>
  File "<frozen importlib._bootstrap>", line 634 in _load_backward_compatible
  File "<frozen importlib._bootstrap>", line 664 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 958 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 969 in _find_and_load
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/site-packages/cx_Freeze/initscripts/Console.py", line 21 in <module>
  File "<frozen importlib._bootstrap>", line 634 in _load_backward_compatible
  File "<frozen importlib._bootstrap>", line 664 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 958 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 969 in _find_and_load
  File "/Users/ultimaker/build/env/3.3/inst/lib/python3.5/site-packages/cx_Freeze/initscripts/__startup__.py", line 12 in <module>
  File "<frozen importlib._bootstrap>", line 634 in _load_backward_compatible
  File "<frozen importlib._bootstrap>", line 664 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 958 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 969 in _find_and_load

**Comment by modem7** [comment]
Looks like it's the same issue as #3734

**Comment by Ellecross** [comment]
@modem7 perhaps it is, but checking both of your logs seem to have a different message. I asked for your config folder in #3734, when you can. :) 

@jimhigson haloo. Can you please attach your 3.2 (roaming) configuration folder here please? I'll try to replicate your issue. Thanks!

**Comment by jimhigson** [comment]
Yes, I can do that. What is the location of those files under OSX?

**Comment by diegopradogesto** [comment]
@jimhigson you can locate the files in `$USER/Library/Application Support/cura/3.3`. Just package this folder into a zip file and attach it to this ticket.

**Comment by jimhigson** [comment]
Have attached my 3.2 and 3.3 settings directories, zipped up.

[3.2.zip](https://github.com/Ultimaker/Cura/files/2024051/3.2.zip)
[3.3.zip](https://github.com/Ultimaker/Cura/files/2024052/3.3.zip)

**Comment by diegopradogesto** [comment]
The problem seems to be one of your custom materials. Loading it in 3.2 I can see this:
![image](https://user-images.githubusercontent.com/15830521/40408805-0ac2e014-5e69-11e8-88c1-7861da206772.png)
Did you manually modify it? The issue is that there is not label in the xml file, but instead `<label />`.
In any case, Cura should not crash, so I will mark it as a bug.

Thanks for the info.

**Comment by diegopradogesto** [comment]
I already did a quick fix for this in b8417bcccbde323159af5dcdf6d5bb29ca70bbf3.
This should not crash in the next release.

**Comment by Ghostkeeper** [comment]
@jimhigson Could you try this with the 3.4 stable release?
